lib/bundling.js

```diff
@@ -0,0 +1,256 @@
+const fs = require('fs');
+const cheerio = require('cheerio');
+const XRegExp = require('xregexp');
+const normalizeSelector = require('normalize-selector');
+const { dirname, basename, resolve, extname, relative } = require('path');
+const { hash } = require('./utils');
+
+
+const cleanWhitespace = (str) => str.replace(/\s+/g, ' ').trim();
+
+
+const removeExtension = (path) => path.replace(/\.beard$/, '');
+
+
+const vdom = (template) => cheerio.load(template, {
+  withDomLvl1: false,
+  normalizeWhitespace: false,
+  xmlMode: false,
+  decodeEntities: false,
+  lowerCaseAttributeNames: false
+});
+
+
+const blockTypes = {
+  handle: {
+    type: 'handle',
+    tag: 'script[handle]',
+    pathsRegex: /(import|require)[^'"`]+['"`]([\.\/][^'"`]+)['"`]/gmi,
+    ext: 'handle.js'
+  },
+  js: {
+    type: 'js',
+    tag: 'script:not(script[handle]):not(script[inline]):not(script[src])',
+    pathsRegex: /(import|require)[^'"`]+['"`]([\.\/][^'"`]+)['"`]/gmi,
+    importStatement: (path) => `import './${path}';`,
+    ext: 'js'
+  },
+  template: {
+    type: 'beard',
+    tag: 'template',
+    ext: 'beard'
+  },
+  css: {
+    type: 'css',
+    tag: 'style:not(style[inline])',
+    pathsRegex: /(@import|url)\s*["'\(]*([^'"\)]+)/gmi,
+    importStatement: (path) => `@import './${path}';`,
+    ext: 'css'
+  }
+};
+
+
+const inlineCSSCommentsRegex = /\/\/[^\n]+/g;
+const combinators = ['>', '+', '~'];
+const deepCombinator = '>>>'; // this is our custom deep combinator for decendant scoping
+const psuedoElements = /::after|:after|::backdrop|::after|:after|::backdrop|::before|:before|::cue|:cue|::first-letter|:first-letter|::first-line|:first-line|::grammar-error|::marker|::part\(.*?\)|::placeholder|::selection|::slotted\(.*?\)|::spelling-error/;
+
+
+function validStyle(val) {
+  if (!val) return false;
+  const notCommentedOut = !val.trim().match(/^\/\//);
+  if (notCommentedOut) return true;
+  return false;
+}
+
+
+function minimizeWhitespace(original$) {
+  const $ = vdom(cleanWhitespace(original$('template').html()));
+  const ommittedTags = 'pre, code, textarea';
+  const originalWhitespaceTags = original$(ommittedTags);
+  const whitespaceTags = $(ommittedTags);
+
+  originalWhitespaceTags.each((i, el) => {
+    $(whitespaceTags[i]).replaceWith(el);
+  });
+
+  return fixInlineAttributeConditions($.html());
+}
+
+
+function addBundleImports(block, blockType) {
+  const { importStatement } = blockType;
+  let { bundle, file } = block;
+
+  bundle = !bundle
```

Comment:
In this context yeah that'd be nice, but in the context of where it comes from no. In case it's not clear, this comes from:

```
<style bundle="account">...</style>
<script bundle="account">...</script>

but in some cases we need a block to be a part of multiple blocks, like this:

<script bundle="base, public">...</script>

To me, it always being plural would be gross since it's the edge case. Also I think it'd be gross having both singular and plural attributes. So to me as is is admittedly not great but better than those two alternatives.
```